toolbox/fdc3-compliance/src/test/fdc3.joinChannel.ts

```diff
@@ -0,0 +1,5 @@
+describe("fdc3.joinChannel", async () => {
+  it("Method is callable", async () => {
+    await window.fdc3.joinChannel("FDC3Conformance"); // This will hang under the Chrome Extension Desktop Agent
```

Comment:
Shouldn't we create the channel first? It isn't a good practice to rely on state from other tests.

Reply:
Do NOT join app channels. This being clarified in the 2.0 docs along with the rename of system channels to app channels. While it's technically possible it's not intended... As such questions have been asked about whether it should be specifically prevented.

You should always be doing getOrCreateChannel then adding a listener to the returned object for a named app channel.

Reply:
Thank you for pointing this out, Kris! This raises a very good question - how can we test `joinChannel()` when we don't have control over the system channels?

Reply:
oof the 1.2 spec doesn't actually require there to be any system channels in the list apart from `global` (which is deprecated - in 1.2 apps MAY be connected to it by default, but don't have to be - its gone in 2.0). 

However, I've not yet seen a desktop agent without system channels and as global is technically required there should always be at least one channel returned by `fdc3.getSystemChannels()` for the app to join. Hence, do that.

Reply:
Thanks for the analysis - we will adopt the suggested use of `fdc3.getSystemChannels()` for this version.